Carbonator/CarbonatorInstance.cs

```diff
@@ -199,7 +199,7 @@ private static void reportMetrics(object state)
                     string metricStr = metric.ToString();
                     // see http://graphite.readthedocs.org/en/latest/feeding-carbon.html
                     Log.Debug("[reportMetrics] reporting: {0}", metricStr.Substring(0, metricStr.Length - 1)); // the Length-1 is to remove the newline at the end for nicer log
-                    byte[] bytes = Encoding.ASCII.GetBytes(metricStr);
+                    byte[] bytes = Encoding.ASCII.GetBytes(conf.Graphite.Prefix + metricStr);
```

Comment:
I'd suggest adding the prefix in ``MetricPathBuilder`` class, inside ``Format`` method, since that is where a metric path string gets generated; instead of appending it in ``reportMetrics``.

You could load the prefix from config in the ``MetricPathBuilder`` constructor that has no arguments (all overloaded constructors call this most basic constructor).

Then in ``Format()`` something like:

```C#
            if (string.IsNullOrEmpty(PathTemplate))
            {
                throw new InvalidOperationException("PathTemplate property is null");
            }
            string template = PathTemplate;
            foreach (string key in Variables.Keys)
            {
                template = template.Replace("%" + key + "%", ReplaceInvalidCharactersInPath(Variables[key]));
            }
            if(!string.IsNullOrEmtpy(Prefix)) 
                template = Prefix + template;
            return template;
```